pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -107,22 +110,22 @@ func (m *Monitor) osdStatus() error {
 }
 
 func (m *Monitor) handleOSDMarkedOut(outOSDid int) error {
-	safeToDestroyOSD, err := client.OsdSafeToDestroy(m.context, m.clusterName, outOSDid)
+	label := fmt.Sprintf("ceph-osd-id=%d", outOSDid)
+	dp, err := k8sutil.GetDeployments(m.context.Clientset, m.clusterName, label)
 	if err != nil {
-		return err
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get osd deployment of osd id %d: %+v", outOSDid, err)
 	}
-
-	if safeToDestroyOSD {
-		logger.Infof("osd.%d is 'safe-to-destroy'", outOSDid)
-		label := fmt.Sprintf("ceph-osd-id=%d", outOSDid)
-		dp, err := k8sutil.GetDeployments(m.context.Clientset, m.clusterName, label)
+	if len(dp.Items) != 0 {
+		safeToDestroyOSD, err := client.OsdSafeToDestroy(m.context, m.clusterName, outOSDid)
 		if err != nil {
-			if errors.IsNotFound(err) {
-				return nil
-			}
-			return fmt.Errorf("failed to get osd deployment of osd id %d: %+v", outOSDid, err)
+			return err
 		}
-		if len(dp.Items) != 0 {
+
+		if safeToDestroyOSD {
+			logger.Infof("osd.%d is 'safe-to-destroy'", outOSDid)
```

Comment:
how about logging immediately before deleting the deployment, inside the next if statement.

Reply:
how about this message:
`osd.%d is 'safe-to-destroy'. removing the osd deployment.`